src/widgets/hits/__tests__/hits.test.tsx

```diff
@@ -0,0 +1,572 @@
+                class="ais-ReverseSnippet-highlighted"
+              >
+                 by Apple.
+              </mark>
+            </span>
+          </p>
+        </li>
+        <li
+          class="ais-Hits-item"
+        >
+          <h2>
+            <span
+              class="ais-Highlight"
+            >
+              <span
+                class="ais-Highlight-nonHighlighted"
+              >
+                Samsung Galaxy 
+              </span>
+              <mark
+                class="ais-Highlight-highlighted"
+              >
+                smartphone
+              </mark>
+            </span>
+          </h2>
+          <h3>
+            <span
+              class="ais-ReverseHighlight"
+            >
+              <mark
+                class="ais-ReverseHighlight-highlighted"
+              >
+                Samsung Galaxy 
+              </mark>
+              <span
+                class="ais-ReverseHighlight-nonHighlighted"
+              >
+                smartphone
+              </span>
+            </span>
+          </h3>
+          <p>
+            <span
+              class="ais-Snippet"
+            >
+              <span
+                class="ais-Snippet-nonHighlighted"
+              >
+                A 
+              </span>
+              <mark
+                class="ais-Snippet-highlighted"
+              >
+                smartphone
+              </mark>
+              <span
+                class="ais-Snippet-nonHighlighted"
+              >
+                 by Samsung.
+              </span>
+            </span>
+          </p>
+          <p>
+            <span
+              class="ais-ReverseSnippet"
+            >
+              <mark
+                class="ais-ReverseSnippet-highlighted"
+              >
+                A 
+              </mark>
+              <span
+                class="ais-ReverseSnippet-nonHighlighted"
+              >
+                smartphone
+              </span>
+              <mark
+                class="ais-ReverseSnippet-highlighted"
+              >
+                 by Samsung.
+              </mark>
+            </span>
+          </p>
+        </li>
+      </ol>
+    </div>
+  </div>
+</div>
+`);
+
+      fireEvent.input(within(searchBoxContainer).getByRole('searchbox'), {
+        target: { value: 'query with no results' },
+      });
+
+      await wait(0);
+
+      expect(container).toMatchInlineSnapshot(`
+<div>
+  <div>
+    <div
+      class="ais-Hits ais-Hits--empty"
+    >
+      <p>
+        No results for 
+        <q>
+          query with no results
+        </q>
+      </p>
+    </div>
+  </div>
+</div>
+`);
+    });
+
+    type CustomHit = { name: string; description: string };
+
+    function createMockedSearchClient(
+      subset: Partial<SearchResponse<CustomHit>> = {}
+    ) {
+      return createSearchClient({
+        search: jest.fn((requests) => {
+          return Promise.resolve(
+            createMultiSearchResponse(
```

Comment:
Regarding the failing type jobs, it seems to originate from the return type of this method. I understand the underlying issue regarding the expected `TObject` type but could not find how to ensure its compatibility with `Hit<CustomHit>`.

In the context of those test cases, we can remove the type inference with `createMultiSearchResponse<any>()` so it stops bugging us, while still maintaining type safety for each single search response below. Wdyt?

Reply:
Yes I agree. The mocks from React InstantSearch Hooks are better typed so instead of trying to fix it here, we can put those in common later on and work arund TypeScript for now.